ios/AzureNotificationHubIOS.js

```diff
@@ -0,0 +1,445 @@
+ * - Add the following to `Link Binary With Libraries`: `libRCTAzureNotificationHubManager.a`
+ * - Add the following to your `Header Search Paths`:
+ * `$(SRCROOT)/../../../node_modules/react-native-azurenotificationhub/ios` and set the search to `recursive`
+ *
+ * Finally, to enable support for `notification` and `register` events you need to augment your AppDelegate.
+ *
+ * At the top of your `AppDelegate.m`:
+ *
+ *   `#import "RCTAzureNotificationHubManager.h"`
+ *
+ * And then in your AppDelegate implementation add the following:
+ *
+ *   ```
+ *    // Required to register for notifications
+ *    - (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings
+ *    {
+ *     [RCTAzureNotificationHubManager didRegisterUserNotificationSettings:notificationSettings];
+ *    }
+ *    // Required for the register event.
+ *    - (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
+ *    {
+ *     [RCTAzureNotificationHubManager didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
+ *    }
+ *    // Required for the registrationError event.
+ *    - (void)application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error
+ *    {
+ *     [RCTAzureNotificationHubManager didFailToRegisterForRemoteNotificationsWithError:error];
+ *    }
+ *    // Required for the notification event.
+ *    - (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)notification
+ *    {
+ *     [RCTAzureNotificationHubManager didReceiveRemoteNotification:notification];
+ *    }
+ *    // Required for the localNotification event.
+ *    - (void)application:(UIApplication *)application didReceiveLocalNotification:(UILocalNotification *)notification
+ *    {
+ *     [RCTAzureNotificationHubManager didReceiveLocalNotification:notification];
+ *    }
+ *   ```
+ */
+class AzureNotificationHubIOS {
+  _data: Object;
+  _alert: string | Object;
+  _sound: string;
+  _badgeCount: number;
+
+  /**
+   * Schedules the localNotification for immediate presentation.
+   *
+   * details is an object containing:
+   *
+   * - `alertBody` : The message displayed in the notification alert.
+   * - `alertAction` : The "action" displayed beneath an actionable notification. Defaults to "view";
+   * - `soundName` : The sound played when the notification is fired (optional).
+   * - `category`  : The category of this notification, required for actionable notifications (optional).
+   * - `userInfo`  : An optional object containing additional notification data.
+   * - `applicationIconBadgeNumber` (optional) : The number to display as the app's icon badge. The default value of this property is 0, which means that no badge is displayed.
+   */
+  static presentLocalNotification(details: Object) {
```

Comment:
Should we have custom logic or should we re-use the existing "presentLocalNotification" features built into PushNotificationIOS in react-native?